### PR TITLE
Fix mismatch in groupselections for RMSD between target and reference

### LINKED ISSUE #2797: Passing an atomgroup to RMSD results in selections within atomgroup
## Expected behavior ##

If I pass an atomgroup to RMSD, `select` and `groupselections` are taken only from that AtomGroup

## Actual behavior ##

At least one of them in at least one of the target or reference does not, resulting in an error when the RMSD is calculated due to mismatching atoms.

## Code to reproduce the behavior ##

Error when selections are not subsets of atomgroups

``` python
In [1]: import MDAnalysis as mda
   ...: from MDAnalysis.tests.datafiles import PSF, DCD, CRD
   ...: from MDAnalysis.analysis import rms

In [2]: u = mda.Universe(PSF, DCD)

In [3]: ca = u.select_atoms('name CA')

In [4]: CORE = 'backbone and (resid 1-29 or resid 60-121 or resid 160-214)'
Help on class RMSD in module MDAnalysis.analysis.rms:

class RMSD(MDAnalysis.analysis.base.AnalysisBase)
   ...: LID = 'backbone and resid 122-159'
   ...: NMP = 'backbone and resid 30-59'

In [5]: R = rms.RMSD(ca, ca, select=CORE,
   ...:              groupselections=[LID, NMP],
   ...:              ref_frame=0)
   ...: R.run()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-76040dc15767> in <module>
      2              groupselections=[LID, NMP],
      3              ref_frame=0)
----> 4 R.run()

~/anaconda3/envs/mdanalysis/lib/python3.7/site-packages/MDAnalysis/analysis/base.py in run(self, start, stop, step, verbose)
    195             self.times[i] = ts.time
    196             # logger.info("--> Doing frame {} of {}".format(i+1, self.n_frames))
--> 197             self._single_frame()
    198         logger.info("Finishing up")
    199         self._conclude()

~/anaconda3/envs/mdanalysis/lib/python3.7/site-packages/MDAnalysis/analysis/rms.py in _single_frame(self)
    663                     refpos, atoms['mobile'].positions,
    664                     weights=self.weights_groupselections[igroup-3],
--> 665                     center=False, superposition=False)
    666         else:
    667             # only calculate RMSD by setting the Rmatrix to None (no need

~/anaconda3/envs/mdanalysis/lib/python3.7/site-packages/MDAnalysis/analysis/rms.py in rmsd(a, b, weights, center, superposition)
    231     N = b.shape[0]
    232     if a.shape != b.shape:
--> 233         raise ValueError('a and b must have same shape')
    234
    235     # superposition only works if structures are centered

ValueError: a and b must have same shape
```

Working when selections are entirely within the atomgroups

``` python
In [7]: bb = u.select_atoms('backbone')

In [8]: R = rms.RMSD(bb, bb, select=CORE,
   ...:              groupselections=[LID, NMP],
   ...:              ref_frame=0)
   ...: R.run()
Out[8]: <MDAnalysis.analysis.rms.RMSD at 0x1183f5f90>
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 1.0.0
- Which version of Python (`python -V`)? 3.7.3
- Which operating system?

### PR DESCRIPTION
**Pull request by 1ut**
Fixes #2797

Changes made in this Pull Request:
 - Edit MDAnalysis.analysis.rms.RMSD to resolve the mismatch in groupselections for RMSD between target and reference.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/rms.py [hunk 1]
```diff
@@ -516,8 +516,8 @@ def __init__(self, atomgroup, reference=None, select='all',
         #   *groupselections* groups each a dict with reference/mobile
         self._groupselections_atoms = [
             {
-                'reference': self.reference.universe.select_atoms(*s['reference']),
-                'mobile': self.atomgroup.universe.select_atoms(*s['mobile']),
+                'reference': self.reference.select_atoms(*s['reference']),
+                'mobile': self.atomgroup.select_atoms(*s['mobile']),
```
Comment (1ut) [inline package/MDAnalysis/analysis/rms.py:520]: I reviewed the codebase and thought It's not essential and wrong to call .universe, so I just removed the character .universe.

Comment (IAlibay) [reply package/MDAnalysis/analysis/rms.py:520]: Going by the docstring, this seems to have been an intentional design decision in order to provide an atom selection that goes beyond the passed atomgroups.

I'll let someone with more familiarity with that feature speak up though.

Comment (lilyminium) [reply package/MDAnalysis/analysis/rms.py:520]: Yes -- this is handy, for example, if you want to align on C-alphas and then calculate the RMSD of backbone atoms.

**Comment by 1ut** [comment]
Close this issue as I have already opened PR #3188.

**Pull request closed without merging**